### PR TITLE
Spare ID safe

### PR DESCRIPTION
**Pull request by DebugOk**
## About the PR
Still a work in progress

TODO:
- [x] Figure out who should get the biscuit
- [X] Get a sprite for the safe
- ☒ ~~Figure out what to put on the card that opens the safe~~ Fuck it, map it
- [x] Get something for the paper background of the cards

## Media

https://github.com/DeltaV-Station/Delta-v/assets/49997488/b1fd56b4-ae88-4ce8-babe-f1c6716dee70

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Added the spare ID safe
- remove: The captain's locker no longer starts with a spare ID card

### CODE REVIEW

File: Content.Server/DeltaV/Biscuit/BiscuitComponent.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+namespace Content.Server.DeltaV.Biscuit;
+
+/// <summary>
+/// This is used for...
+/// </summary>
+[RegisterComponent]
+public sealed partial class BiscuitComponent : SharedBiscuitComponent
+{
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("cracked")]
```
Comment (deltanedas) [inline Content.Server/DeltaV/Biscuit/BiscuitComponent.cs]: ("cracked") not required + vvrw is implied for datafields whenever you update engine, so you can remove that too after done with all the testing of it

File: Content.Server/Paper/PaperComponent.cs [hunk 2]
```diff
@@ -16,6 +16,10 @@ public sealed partial class PaperComponent : SharedPaperComponent
     [DataField("stampedBy")]
     public List<StampDisplayInfo> StampedBy { get; set; } = new();
 
+    [DataField("canEdit")]
```
Comment (deltanedas) [inline Content.Server/Paper/PaperComponent.cs:19]: same

File: Content.Server/DeltaV/Biscuit/BiscuitComponent.cs (hunk 1 above)
Comment (DebugOk) [reply Content.Server/DeltaV/Biscuit/BiscuitComponent.cs]: They get VVR by default, RW needs to be explicitly defined iirc.

File: Content.Server/Paper/PaperComponent.cs (hunk 2 above)
Comment (DebugOk) [reply Content.Server/Paper/PaperComponent.cs:19]: This keeps it consistent with the rest of the file

File: Content.Server/DeltaV/Biscuit/BiscuitComponent.cs [hunk 3]
```diff
@@ -8,7 +8,5 @@ namespace Content.Server.DeltaV.Biscuit;
 [RegisterComponent]
 public sealed partial class BiscuitComponent : SharedBiscuitComponent
 {
-    [ViewVariables(VVAccess.ReadWrite)]
-    [DataField("cracked")]
```
Comment (deltanedas) [inline Content.Server/DeltaV/Biscuit/BiscuitComponent.cs]: pro still need [DataField] just the tag is optional :trollface:

Comment (DebugOk) [reply Content.Server/DeltaV/Biscuit/BiscuitComponent.cs]: Me when doing several things at once backfires horribly

**Comment by DebugOk** [comment]
It may be best to map the biscuit in a place you dont need any access to get to, in case it happens there is no command at all

File: Resources/Prototypes/DeltaV/Entities/Objects/Misc/paperslips.yml [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            - !type:DoActsBehavior
+              acts: [ "Destruction" ]
+
+- type: entity
+  parent: PaperSlip
+  id: PaperSlipCorporate
+  name: corporate plastic card
+  description: A plastic card for confidential corporate matters.
+  components:
+    - type: Damageable
```
Comment (deltanedas) [inline Resources/Prototypes/DeltaV/Entities/Objects/Misc/paperslips.yml]: this is inherited from the parent remove 2 lines

**Review by deltanedas — COMMENTED** [review]
pro

**Comment by deltanedas** [comment]
apparently in tg i think the biscuit spawns if there is no cap then the heads/engis rush to get aa
will it have some spawn logic or be mapped in vault

**Comment by DebugOk** [comment]
> apparently in tg i think the biscuit spawns if there is no cap then the heads/engis rush to get aa will it have some spawn logic or be mapped in vault

In TG, it'll try to give it to a random command member if there is no captain present. If theres no command, it'll go to any random crewmember iirc

I dont really feel like making any logic for it since I have no idea where to begin with that, so I'm still deciding between mapped in and in a locker

**Comment by deltanedas** [comment]
if its in locker you have to break it open to get it which defeats the point :trollface:

**Comment by DebugOk** [comment]
@leonardo-dabepis pls review sprites

**Comment by leonardo-dabepis** [comment]
> @leonardo-dabepis pls review sprites

This all looks good, I especially like the id card safe 👍

**Comment by DebugOk** [comment]
> > @leonardo-dabepis pls review sprites
> 
> This all looks good, I especially like the id card safe 👍

Yeah thats the only unique one, the rest either came from TG or got cobbled together from tg stuff

File: Resources/Prototypes/DeltaV/Entities/Objects/Specific/Command/safe.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+﻿- type: entity
+  parent: FireAxeCabinet
+  id: SpareIdCabinet
+  name: spare id cabinet
+  description: There is a small label that reads "For Emergency use only".
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/DeltaV/Entities/Objects/Specific/Command/safe.yml]:
```suggestion
  description: There is a small label that reads "For emergency use only".
```

**Review by DEATHB4DEFEAT — COMMENTED** [review]
Pretty cool

**Review by DEATHB4DEFEAT — APPROVED** [review]

**Comment by DebugOk** [comment]
We ball. Me omw to make the captain's card unavailable until sop is altered

**Pull request merged**